agimus_controller_ros/agimus_controller_ros/agimus_controller.py

```diff
@@ -81,68 +122,99 @@ def initialize_ros_attributes(self) -> None:
                 reliability=ReliabilityPolicy.BEST_EFFORT,
             ),
         )
+        self.state_subscriber = self.create_subscription(
+            JointState,
+            "joint_states",
+            self.joint_states_callback,
+            qos_profile=QoSProfile(
+                depth=10,
+                reliability=ReliabilityPolicy.BEST_EFFORT,
+            ),
+        )
         if self.params.publish_debug_data:
             self.ocp_solve_time_pub = self.create_publisher(
-                builtin_interfaces.msg.Duration, "ocp_solve_time", 10
+                builtin_interfaces.msg.Duration,
+                "ocp_solve_time",
+                qos_profile=QoSProfile(
+                    depth=10,
+                    reliability=ReliabilityPolicy.BEST_EFFORT,
+                ),
+            )
+            self.ocp_x0_pub = self.create_publisher(
+                Sensor,
+                "ocp_x0",
+                qos_profile=QoSProfile(
+                    depth=10,
+                    reliability=ReliabilityPolicy.BEST_EFFORT,
+                ),
             )
-            self.ocp_x0_pub = self.create_publisher(Sensor, "ocp_x0", 10)
         self.create_timer(1.0 / self.params.rate, self.run_callback)
 
+    def setup_mpc(self):
+        """Creates mpc, ocp, warmstart"""
+
+        ocp_params = OCPParamsBaseCroco(
+            dt=self.params.ocp.dt,
+            horizon_size=self.params.ocp.horizon_size,
+            solver_iters=self.params.ocp.max_iter,
+            callbacks=self.params.ocp.activate_callback,
+            qp_iters=self.params.ocp.max_qp_iter,
+            dt_factor_n_seq=[(1, self.params.ocp.horizon_size)],
```

Comment:
I dont think this parameters should be in OPCParams since OCP does not have and does not handle the buffer, it is taken care of from outside

maybe it can be in YAML

Reply:
okay, @MedericFourmy pointed out that dt also needs to be applied in OCP, but I don't see it as a part of merged PR. Does anyone work on implementing it?

Reply:
This part of the code is worked on by @TheoMF and @MaximilienNaveau